eisbuk-admin/cypress/plugins/firebasePlugin/updateFirestore/readTestData.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-catch-shadow */
+import * as fs from "fs";
+import * as path from "path";
+
+import { FirestoreDataUpdate } from "../types";
+
+/**
+ * A utility function that reads the provided test file from the provided
+ * test data directory. After reading the file, returns the expected collections
+ * (if any) in form of `{[collection]: collectionData}` object
+ * @param dir absolute path to `__testData__` directory
+ * @param fName name of the test data file to read and parse
+ * @returns `{[collection]: collectionData}` object
+ */
+const readTestData = async (
+  dir: string,
+  files: string[]
+): Promise<FirestoreDataUpdate> => {
+  // read and parse the files
+  const parsedFiles = await Promise.all(
+    files.map((fName) => {
+      // trim ".json" (if included in the filename)
+      // and add ".json" at the end (to ensure the files has only one ".json" extension)
+      const jsonFName = fName.replace(".json", "") + ".json";
```

Comment:
Is it necessary to support two different ways to refer to the same file?
If it really is, this implementation causes a problem with

```javascript
"my-files.json/aaa.json".replace(
  ".json", "") + ".json" === 'my-files/aaa.json.json'
```

To fix this we can specify we need `.json` to be at the end of the string:

```javascript
"my-files.json/aaa.json".replace(
  /\.json$/, "") + ".json" === 'my-files.json/aaa.json'
```
or no automatic appending of `.json` (my favourite).

Reply:
Right...I didn't really account for that scenario...and ok, I'll do it without the name cleanup..I guess there are a few equally "right" ways to go about it and if you prefer that...I don't have a preference.